website/docs/r/dashboard.html.markdown

```diff
@@ -0,0 +1,824 @@
+            aggregation = "count"
+            facet = "@duration"
+            interval = 5000
+          }
+          search = {
+            query = "status:info"
+          }
+          group_by {
+            facet = "host"
+            limit = 10
+            sort = {
+              aggregation = "avg"
+              order = "desc"
+              facet = "@duration"
+            }
+          }
+        }
+        display_type = "area"
+      }
+      request {
+        apm_query {
+          index = "apm-search"
+          compute = {
+            aggregation = "count"
+            facet = "@duration"
+            interval = 5000
+          }
+          search = {
+            query = "type:web"
+          }
+          group_by {
+            facet = "resource_name"
+            limit = 50
+            sort = {
+              aggregation = "avg"
+              order = "desc"
+              facet = "@string_query.interval"
+            }
+          }
+        }
+        display_type = "bars"
+      }
+      request {
+        process_query {
+          metric = "process.stat.cpu.total_pct"
+          search_by = "error"
+          filter_by = ["active"]
+          limit = 50
+        }
+        display_type = "area"
+      }
+      marker {
+        display_type = "error dashed"
+        label = " z=6 "
+        value = "y = 4"
+      }
+      marker {
+        display_type = "ok solid"
+        value = "10 < y < 999"
+        label = " x=8 "
+      }
+      title = "Widget Title"
+      title_size = 16
+      title_align = "left"
+      time = {
+        live_span = "1h"
+      }
+    }
+  }
+
+  widget {
+    toplist_definition {
+      request {
+        q= "avg:system.cpu.user{app:general} by {env}"
+        conditional_formats {
+          comparator = "<"
+          value = "2"
+          palette = "white_on_green"
+        }
+        conditional_formats {
+          comparator = ">"
+          value = "2.2"
+          palette = "white_on_red"
+        }
+        }
+      title = "Widget Title"
+      title_size = 16
+      title_align = "left"
+      time = {
+        live_span = "1h"
+      }
+    }
+  }
+
+  widget {
+    group_definition {
+      layout_type = "ordered"
+      title = "Group Widget"
+
+      widget {
+        note_definition {
+          content = "cluster note widget"
+              background_color = "yellow"
+        }
+      }
+      widget {
+        alert_graph_definition {
+          alert_id = "123"
+          viz_type = "toplist"
+          title = "Alert Graph"
+          title_size = "16"
+          title_align = "right"
+          time = {
+            live_span = "1h"
+          }
+        }
+      }
+    }
+  }
+
+  template_variable {
+		name   = "var_1"
+		prefix = "host"
+		default = "aws"
+	}
+	template_variable {
+		name   = "var_2"
+		prefix = "service_name"
+		default = "autoscaling"
+	}
+}
+
+# Create a new Datadog dashboard - Free layout
+resource "datadog_dashboard" "free_dashboard" {
```

Comment:
Indentation in this example is a little weird as well, let's keep using two spaces for each level of indentation.